### PR TITLE
🛡️ Sentinel: [security improvement] Harden scripts against option injection

### PR DESCRIPTION
**Pull request by d-o-hub**
🚨 Severity: MEDIUM
💡 Vulnerability: Option injection in shell scripts via unvalidated file paths or patterns.
🎯 Impact: Files starting with a hyphen could be interpreted as command-line options by tools like grep, sed, or chmod, leading to script failures or potential misuse.
🔧 Fix: Added -- delimiter to chmod, grep, and sed calls. Used -e in grep and sed to explicitly define patterns/scripts.
✅ Verification: Verified with manual tests against files named with hyphens and ran existing security tests. Quality gate script confirmed no regressions in validation logic.

---
*PR created automatically by Jules for task [5584381877453903793](https://jules.google.com/task/5584381877453903793) started by @d-o-hub*

### CODE REVIEW

File: scripts/propagate-version.sh [hunk 1]
```diff
@@ -48,38 +48,38 @@ for file in "${FILES_TO_UPDATE[@]}"; do
     fi
 
     # Update version badges: version-0.X.Y -> version-NEW_VERSION
-    if grep -q "version-[0-9]" "$file" 2>/dev/null; then
-        sed -i "s/version-[0-9]\+\.[0-9]\+\.[0-9]\+/version-${VERSION}/g" "$file"
+    if grep -q -e "version-[0-9]" -- "$file" 2>/dev/null; then
+        sed -i -e "s/version-[0-9]\+\.[0-9]\+\.[0-9]\+/version-${VERSION}/g" -- "$file"
```
[the remark this answers is not in the document]
Comment (d-o-hub) [reply scripts/propagate-version.sh]: @jules address feedback, git commit, git push all changes

**Comment by d-o-hub** [comment]
> ### Pull Request Overview
> The PR successfully identifies several shell injection vectors, but the current solution introduces regressions for macOS/BSD users. Specifically, the hardening of `sed` commands in `scripts/propagate-version.sh` relies on GNU-specific behavior for the `-i` flag and the `--` delimiter, which will cause execution failures on BSD-based systems (including macOS).
> 
> While Codacy reports the PR as 'up to standards', the lack of automated test scenarios for the specific injection cases (e.g., handling files named `-h`) makes it difficult to verify the effectiveness of the hardening. Furthermore, there are gaps in implementation where some `grep` calls remain unhardened. These issues should be addressed to ensure the scripts are both secure and portable across environments.
> 
> #### About this PR
> * The PR lacks automated test cases to verify the fix against actual option injection scenarios (e.g., attempting to process files named '-h'). Relying on existing tests or manual verification may not catch edge cases introduced by the new flags.
> 
> `1` comment outside of the diff
> #### Test suggestions
> * [ ]  Verify chmod correctly handles files with names starting with a hyphen (e.g., -filename)
> * [ ]  Verify grep treats patterns starting with a hyphen as literal strings when -e is provided
> * [ ]  Verify sed treats filenames starting with a hyphen as paths when -- is provided
> * [ ]  Verify version propagation logic remains functional after the addition of security flags
> 
> Prompt proposal for missing tests
> Low confidence findings
> `TIP` Improve review quality by [adding custom instructions](https://docs.codacy.com/codacy-ai/codacy-ai/#custom-instructions) `TIP` How was this review? [Give us feedback](https://tally.so/r/jaBlA1?org=d-o-hub&repo=github-template-ai-agents&pr=374)

@jules address feedback, git commit, git push all changes
